packages/react/src/hooks/useVisibilityLoader.js

```diff
@@ -0,0 +1,75 @@
+import { useEffect, useLayoutEffect, useState } from 'react';
+import warning from 'warning';
+
+import { browserSupports } from '../utils/componentUtilityFunctions';
+
+const useVisibilityLoader = (ref, { maxToLoad = 1, loadBatchSize = 20 } = {}) => {
+  if (!browserSupports('IntersectionObserver')) {
+    warning(
+      !__DEV__,
+      'The current browser does not support ResizeObserver. You will need to include a ResizeObserver polyfill for this component to function properly.'
+    );
+    return [true, false];
+  }
+
+  /* eslint-disable react-hooks/rules-of-hooks */
+  const [isVisible, setIsVisible] = useState(maxToLoad > 1 ? [] : false);
+  const [moreToLoad, setMoreToLoad] = useState(true);
+  const [nextVisibleIndex, setNextVisibleIndex] = useState(0);
+
+  useEffect(() => {
+    setMoreToLoad(nextVisibleIndex < maxToLoad);
+  }, [maxToLoad, nextVisibleIndex]);
+
+  useLayoutEffect(() => {
+    if (ref.current && moreToLoad) {
+      const observer = new IntersectionObserver(
```

Comment:
I think this will create a new intersection observer with each render. We may want to store the observer in a ref 
in order to clean up on each render.

Reply:
Done.